services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -0,0 +1,56 @@
+package cromwell.pipeline.service
+import java.nio.file.Path
+
+import com.typesafe.config.{ Config, ConfigFactory }
+import cromwell.pipeline.datastorage.dto.{ Project, ProjectFile, Version }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GitLabProjectVersioning(httpClient: HttpClient)(
+  implicit executionContext: ExecutionContext
+) extends ProjectVersioning[VersioningException] {
+  import GitLabConfig._
+
+  override def updateFile(project: Project, projectFile: ProjectFile): AsyncResult[String] = ???
+  override def updateFiles(project: Project, projectFiles: ProjectFiles): AsyncResult[List[String]] = ???
+
+  override def createRepository(project: Project): AsyncResult[Project] = {
+    def responseFuture = httpClient.post(URL + "projects", createRepositoryParams(project), TOKEN, "")
+    if (!project.active)
+      Future.failed(VersioningException("Could not create a repository for not an active project."))
+    else
+      responseFuture.flatMap(
+        resp =>
+          Future.successful(
+            if (resp.status != 201)
+              Left(VersioningException(s"The repository was not created. Response status: ${resp.status}"))
+            else Right(updateProject(project))
+          )
+      )
+  }
+
+  override def getFiles(project: Project, path: Path): AsyncResult[List[String]] = ???
+  override def getProjectVersions(project: Project): AsyncResult[Project] = ???
+  override def getFileVersions(project: Project, path: Path): AsyncResult[List[Version]] = ???
+  override def getFilesVersions(project: Project, path: Path): AsyncResult[List[Version]] = ???
+  override def getFileTree(project: Project, version: Option[Version]): AsyncResult[List[String]] = ???
+  override def getFile(project: Project, path: Path, version: Option[Version]): AsyncResult[String] = ???
```

Comment:
Maybe we need a GitLabProjectVersioning trait and class implementation as addition(in this PR), not to make columns of unimplemented methods? It seems strange to me. Maybe it would better to make this class just abstract?

Reply:
We already have tasks for all these methods to implement.

Reply:
It's a lot of work to implement whole class. So I guess we can divided it.
And to make project compilable during this time stubbing methods with `???` is ok.